int32 int32In
---
int32 int32Out